project_earned_value_task/models/project.py

```diff
@@ -0,0 +1,472 @@
+    _inherit = 'project.task.type'
+    poc = fields.Float('% Completed')
+
+
+class Project(models.Model):
+    
+    _inherit = "project.project"
+
+    @staticmethod
+    def _get_evm_ratios(ac, pv, ev, bac):
+
+        res = {
+            'ac': ac,
+            'pv': pv,
+            'ev': ev,
+            'bac': bac,
+        }
+
+        # PCC: Costs to date / Total costs
+        try:
+            res['pcc'] = res['ac'] / res['pv']
+        except ZeroDivisionError:
+            res['pcc'] = 0
+
+        # SV: Schedule variance
+        res['sv'] = res['ev'] - res['pv']
+
+        # SVP: Schedule variance in percentage
+        try:
+            res['svp'] = res['sv'] / res['pv']
+        except ZeroDivisionError:
+            res['svp'] = 0
+
+        # SPI: Schedule Performance Index
+        try:
+            res['spi'] = res['ev'] / res['pv']
+        except ZeroDivisionError:
+            res['spi'] = 0
+
+        # CV: Cost Variance
+        res['cv'] = res['ev'] - res['ac']
+
+        # CVP: Cost Variance Percent
+        try:
+            res['cvp'] = (res['cv'] / res['ev']) * 100
+        except ZeroDivisionError:
+            res['cvp'] = 0
+
+        # CPI: Cost Performance Index
+        try:
+            res['cpi'] = res['ev'] / res['ac']
+        except ZeroDivisionError:
+            res['cpi'] = 1
+
+        # TCPI: To-complete Performance Index
+        bac_ac_amount = res['bac'] - res['ac']
+        try:
+            res['tcpi'] = (res['bac'] - res['ev']) / bac_ac_amount
+        except ZeroDivisionError:
+            res['tcpi'] = 1
+
+        # EAC: Estimate at completion
+        try:
+            res['eac'] = res['bac'] / res['cpi']
+        except ZeroDivisionError:
+            res['eac'] = res['bac']
+
+        # VAC: Variance at Completion
+        res['vac'] = res['bac'] - res['eac']
+
+        # VACP: Variance at Completion Percent
+        try:
+            res['vacp'] = (res['vac'] / res['bac']) * 100
+        except ZeroDivisionError:
+            res['vacp'] = 0
+
+        # ETC: Estimate To Complete
+        try:
+            res['etc'] = (res['bac'] - res['ev']) / res['cpi']
+        except ZeroDivisionError:
+            res['etc'] = 0
+
+        # EAC: Estimate At Completion
+        try:
+            res['eac'] = res['bac'] / res['cpi']
+        except ZeroDivisionError:
+            res['eac'] = 0
+
+        # POC - Percent of Completion
+        try:
+            res['poc'] = res['ev'] / res['bac'] * 100
+        except ZeroDivisionError:
+            res['poc'] = 0
+
+        return res
+
+    def _get_earliest_latest_dates(self):
+        project = self
+        cr = self.env.cr
+        # Get the earliest and latest dates for the associated tasks
+        cr.execute("""SELECT MIN(date_start)
+                   FROM project_task
+                   WHERE project_id=%s""", (project.id, ))
+        min_date_start = cr.fetchone()[0] or 0
+        if min_date_start == 0:
```

Comment:
Unless you want to specifically exclude `None`values, it's more pythonic to write `if min_date_start:`